### ISSUE TITLE
Version 5.0.10 Doesn't read contacts despite rights given

### ISSUE THREAD

**Issue by devendbhatt** [signalapp/Signal-Android#10596]
- [ ] I have searched open and closed issues for duplicates
- [ ] I am submitting a bug report for existing functionality that does not work as intended
- [ ] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [ ] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Signal App unable / fails to read contacts saying "Error retrieving contacts, check your network connection."
### Steps to reproduce
- Install Signal version 5.0.10 or 5.1.8 from Google Play Store. 
- During first run of the app, allow all rights it asks for. 
- After completing all required steps during first run, try to create a chat with a friend and the apps goes to list all your contacsa who have installed Sifnal app and it fails to list the same.  

**Actual result:** Signal App can't list out contacts to choose from to start chat with.  
**Expected result:** Signal App should list out contacts to choose from to start chat with
### Device info

**Device:** vivo V7 +
**Android version:** 8.1
**Signal version:** 5.0.10

### Link to debug log
https://debuglogs.org/bcf413e11f0b6d3684b7f49f0a9f1f82410ed664308f841a54ff0bf7dfed3f1a

**Comment by lgschell** [comment]
Hi, I've observed the same issue as described above. I've installed Signal version 5.0.10 and not all the contacts were listed out. Then, I've updated Signal to version 5.1.8 with the same outcome. Thank you.

I've also tried below actions without success:

a- Removed Signal account from Android Settings
b- Refresh Contacts from Signal
c- Restarted the mobile
d- On Signal, manually entered a mobile number ( from another Signal user with v5.1.8 ) but Signal didn't recognised the number as a Signal one

Device info

Device: Samsung Galaxy A21s
Android version: 10
Signal version: 5.1.8
